src/dataset/impl_dataset.rs

```diff
@@ -247,13 +247,122 @@ impl<F: Float, D: Data<Elem = F>, I: Dimension> From<ArrayBase<D, I>>
 }
 
 impl<F: Float, T: Targets, D: Data<Elem = F>, I: Dimension> From<(ArrayBase<D, I>, T)>
-    for Dataset<ArrayBase<D, I>, T>
+    for DatasetBase<ArrayBase<D, I>, T>
 {
     fn from(rec_tar: (ArrayBase<D, I>, T)) -> Self {
-        Dataset {
+        DatasetBase {
             records: rec_tar.0,
             targets: rec_tar.1,
             weights: Vec::new(),
         }
     }
 }
+
+impl<F: Float, E: Clone> Dataset<F, E> {
+    pub fn map_targets_array<T, G: FnMut(&E) -> T>(self, fnc: G) -> Dataset<F, T> {
+        let DatasetBase {
+            records,
+            targets,
+            weights,
+            ..
+        } = self;
+
+        let new_targets = targets.iter().map(fnc).collect::<Vec<T>>();
+        let new_targets = Array1::from_shape_vec(new_targets.len(), new_targets).unwrap();
+
+        DatasetBase {
+            records,
+            targets: new_targets,
+            weights,
+        }
+    }
```

Comment:
I had to add it in order to change the return type of functions in the `datasets` sub-crate, since `map_targets` puts the new targets in a `Vec`